python-package/lightgbm/compat.py

```diff
@@ -164,7 +164,17 @@ class _LGBMRegressorBase:  # type: ignore
     from dask.distributed import Client, Future, default_client, wait
 
     DASK_INSTALLED = True
-except ImportError:
+# catching 'ValueError' here because of this:
+# https://github.com/microsoft/LightGBM/issues/6365#issuecomment-2002330003
+#
+# That's potentially risky as dask does some significant import-time processing,
+# like loading configuration from environment variables and files, and catching
+# ValueError here might hide issues with that config-loading.
+#
+# But in exchange, it's less likely that 'import lightgbm' will fail for
+# dask-related reasons, which is beneficial for any workloads that are using
+# lightgbm but not its Dask functionality.
+except (ImportError, ValueError):
```

Comment:
In the future, we might catch `ValueError` and re-raise it if the error message is not "Must install dask-expr to activate query planning". Obviously, this isn't ideal either as `dask` may change the message at any time...

Reply:
In any case, reading through #6365, I find it surprising that the `ValueError` is raised at all. If `dask` is a non-installed optional dependency, the `ImportError` should be raised first. If it is installed, it does not seem to be an issue, at least, for `conda-forge` -- is the `pip` package for `dask` not configured correctly? 🤔

Reply:
> re-raise it if the error message is not "Must install dask-expr to activate query planning"

I'd rather not take on any coupling to specific error messages. I hope my PR to `dask` will be accepted, and then we could remove this addition of `ValueError`.

> If dask is a non-installed optional dependency, the ImportError should be raised first

The `ValueError` doesn't happen at *install* time of `dask`. It happens at *import* time.

If you haven't yet, see the description and diff at https://github.com/dask/dask/pull/11007. This `ValueError` comes from code that only runs when you try to import `dask.dataframe`.

>  is the `pip` package for dask not configured correctly/

The `dask` package on conda-forge lists `dask-expr` as a strong runtime dependency ([code link](https://github.com/conda-forge/dask-feedstock/blob/d16480dc8abea2ee33900604672ac85e52ef8fb1/recipe/meta.yaml#L29)).

The `dask` wheel on PyPI does not... but it does list `dask-expr` as a dependency of `dask[dataframe]` ([code link](https://github.com/dask/dask/blob/f9310c440b661f1ae96ffd8726404d3c3fe32edc/pyproject.toml#L58)).

Neither of those is "correct" or "incorrect"... they're just different.